stan-dev/math
# Binary vectorization and testing framework

#### Submission Checklist


#### Summary:
This pull request adds code for binary vectorization and creates a testing framework to test it.

#### Intended Effect:
This will add code so that functions that take two arguments can be applied to matrix and vector arguments. It also adds code to test functions that have been vectorized. However, the code being added does not support broadcasting.

#### How to Verify:
Run `./runTests.py test/unit/math/mix/mat/vectorize/binary_foo_test.cpp` and make sure that all tests passes. The example function, `binary_foo`, is in `test/unit/math/prim/mat/vectorize/binary_foo.hpp`.

#### Side Effects:

#### Documentation:
N/A. Code is documented, but this change doesn't affect users.

#### Copyright and Licensing
Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Rayleigh Lei owns the copyright to this work.

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)
